math/octave-devel/Portfile

```diff
@@ -171,5 +171,97 @@ configure.args-append       --disable-silent-rules                  \
                             --disable-java                          \
                             --with-qt=4
 
-                            
+variant app description "build application bundle to launch octave-devel" {
+    depends_build-append port:librsvg
+
+    if {[vercmp ${xcodeversion} 4.5] < 0} {
+        # need a way to generate icns file for XCode prior to 4.5
+        # see #51487
+        depends_build-append port:libicns
+    }
+
+    global appName
+    set appName Octave_${version}.app
+
+    post-build {
+        xinstall -d -m 0755 ${worksrcpath}/${appName}/Contents
+        xinstall -d -m 0755 ${worksrcpath}/${appName}/Contents/Resources
+        xinstall -d -m 0755 ${worksrcpath}/${appName}/Contents/MacOS
+
+        set script [open "${worksrcpath}/${appName}/Contents/MacOS/Octave" w 0755]
+        if { [variant_isset qt4] || [variant_isset qt5] } {
+            puts ${script} "#!/bin/sh"
+            puts ${script} ""
+            puts ${script} "${prefix}/bin/octave-${version} --force-gui"
+        } else {
+            puts ${script} "#!/usr/bin/osascript"
+            puts ${script} ""
+            puts ${script} "tell application \"Terminal\" to do script \"${prefix}/bin/octave-${version} --no-gui-libs; exit\""
+        }
```

Comment:
I think ```octave-${version}``` depended on the previous octave-devel's destrooting behaviour.

Reply:
The version is properly set, but will lead to a rather ugly App name...

Reply:
doesn't ${version} expand to something like 4.3.0+_20171102 ?

Reply:
That's what I meant by ugly! Something shorter like Octave-4.3.0+ might look better.